src/libPMacc/include/eventSystem/events/kernelEvents.hpp

```diff
@@ -50,6 +50,7 @@ namespace PMacc
         CUDA_CHECK_KERNEL_MSG(cudaThreadSynchronize(),"Crash after kernel call");       \
         taskKernel->activateChecks();                                                   \
         CUDA_CHECK_KERNEL_MSG(cudaThreadSynchronize(),"Crash after kernel activation"); \
+        CUDA_CHECK( cudaGetLastError( ) );
```

Comment:
Please use `CUDA_CHECK_KERNEL_MSG` and move this check before line 50.
Than this call checks the asynchronous cuda launch and `cudaThreadSynchronize()` that the kernel end without errors.